department-of-veterans-affairs/vets-api
# [9806] Adding ability for MPI to create a new user

## Description of change
This PR adds the ability for the sign in service to create a user in MPI if it does not already exist. If a user is verified, but does not have a retrievable icn, then it is determined that they don't have an MPI entry, and a new record is created

## Original issue(s)
department-of-veterans-affairs/vets-api#9806

## Things to know about this PR
- Testing on localhost is going to be weird, with the fact that MPI is mocked. This might be an example of a commit we actually have to test on a review instance
- On an environment where MPI isn't mocked, you should log in with a new verified id.me or login.gov record, one you explicitly haven't logged in with eauth before
- After logging in with a new account, you should be able to call the `/introspect` route and see an ICN





I was able to test this on a review instance:
- Log into a review instance rails console
- Create a user and an identity object
`identity = UserIdentity.new(first_name: 'Billy', last_name: 'Billyngton', ssn: '384756394', birth_date: '10-10-2020', logingov_uuid: SecureRandom.uuid, uuid: SecureRandom.uuid, loa: {current: 3, highest: 3})`
`user = User.new`
`user.instance_variable_set(:@identity)`
- Confirm that user is `loa3` without an `icn`
`user.loa3?` => `true`
`user.icn` => `nil`
- Create a new MPI record for user:
`user.mpi_add_person_implicit_search`
- Now attempt to retrieve `icn` from user again:
`user.icn` => `1013677553V120625`
- You can confirm an MPI record exists for user with:
`user.send(:mpi).profile`
